pkg/e2e/framework/machinehealthcheck.go

```diff
@@ -118,3 +121,29 @@ func StopKubelet(nodeName string) error {
 	}
 	return client.Create(context.TODO(), pod)
 }
+
+// CreateOrUpdateTechPreviewFeatureGate creates or updates if it already exists the cluster feature gate with tech preview features
+func CreateOrUpdateTechPreviewFeatureGate() error {
+	client, err := LoadClient()
+	if err != nil {
+		return err
+	}
+
+	fg := &osconfigv1.FeatureGate{}
+	key := types.NamespacedName{
+		Name:      operator.MachineAPIFeatureGateName,
+		Namespace: TestContext.MachineApiNamespace,
+	}
+	err = client.Get(context.TODO(), key, fg)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			fg.Name = operator.MachineAPIFeatureGateName
+			fg.Namespace = TestContext.MachineApiNamespace
+			fg.Spec = osconfigv1.FeatureGateSpec{FeatureSet: osconfigv1.TechPreviewNoUpgrade}
+			return client.Create(context.TODO(), fg)
+		}
+		return err
+	}
+	fg.Spec = osconfigv1.FeatureGateSpec{FeatureSet: osconfigv1.TechPreviewNoUpgrade}
```

Comment:
You might also check if `osconfigv1.FeatureGateSpec.FeatureSet` is already set to `osconfigv1.TechPreviewNoUpgrade` and `return nil`.

Reply:
done